src/field.rs

```diff
@@ -143,10 +144,22 @@ macro_rules! make_field {
 
         impl PartialEq for $elem {
             fn eq(&self, rhs: &Self) -> bool {
+                // The fields included in this comparison MUST match the fields
+                // used in Hash::hash
+                // https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq
                 $fp.from_elem(self.0) == $fp.from_elem(rhs.0)
             }
         }
 
```

Comment:
What's going on here? Why do we need to implement this?

Reply:
As noted in the docs on trait `Hash`, it must hold that `k1 == k2 -> hash(k1) == hash(k2)` or bad things will happen if a `Field` is ever used as a key in a `HashMap`. IIUC, the derived `Hash` implementation that I replaced with the new impl would do `self.0.hash(state)`, meaning it wasn't hashing the value modulo the field prime, and this would violate the relation between `Eq` and `Hash` that Rust wants us to enforce.

The lint in question:  https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq

Reply:
Thanks for clarifying.